src/net/atlas/download.rs

```diff
@@ -719,6 +719,20 @@ impl<T: Ord + Requestable + fmt::Display + std::hash::Hash> BatchedRequestsState
                     return fsm;
                 }
 
+                let _ = PeerNetwork::with_network_state(
+                    network,
+                    |ref mut network, ref mut network_state| {
+                        for (event_id, peer) in state.faulty_peers.drain() {
+                            debug!(
+                                "Atlas: Deregistering faulty connection (event_id: {}, peer: {})",
+                                event_id, peer
+                            );
+                            network.http.deregister_http(network_state, event_id);
```

Comment:
It would be better if you simply returned the list of event IDs to the PeerNetwork state machine, so we can keep all the deregistrations in one place.  The block downloader does this, for example.

Reply:
Addressed with latest